Dart/src/com/jetbrains/lang/dart/Dart.bnf

```diff
@@ -382,7 +382,7 @@ formalParameterList ::= '(' ')' |
                         '(' optionalFormalParameters ')'
 private normalFormalParameters ::= normalFormalParameter (',' normalFormalParameter)*
 
-fieldFormalParameter ::= metadata* finalConstVarVoidOrType? 'this' '.' referenceExpression typeParameters? formalParameterList?
+fieldFormalParameter ::= metadata* finalConstVarVoidOrType? ('this' | 'super') '.' referenceExpression typeParameters? formalParameterList?
```

Comment:
You might want to rename the production to `fieldOrSuperFormalParameter`, but it doesn't change the semantics, so up to you.